lib/models/measurement.js

```diff
@@ -18,9 +18,15 @@ const measurementSchema = new Schema({
     type: Date,
     required: true,
     default: moment.utc().toDate()
+  },
```

Comment:
7a9b2379
I removed the separate collection, as the references were making things indeed very complicated.

By storing the location history directly in `box.locations` (and `measurement.location`), response times were cut in half on average!
This might be further improved by adding the correct DB index.